marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/builtins/steps/mastering/default/merging.sjs

```diff
@@ -22,13 +22,24 @@ const processedURIs = [];
 
 function main(content, options) {
   // These index references can't be out this function scope or the jobReport will error, since they don't exist for the jobs DB
+  if (options.stepId) {
+    const stepDoc = fn.head(cts.search(cts.andQuery([
+      cts.collectionQuery("http://marklogic.com/data-hub/steps"),
+      cts.jsonPropertyValueQuery("stepId", options.stepId, "case-insensitive")
+    ])));
+    if (stepDoc) {
+      options = stepDoc.toObject();
+      isSeparateMatchStep = true;
+    } else {
+      fn.error(null, 'RESTAPI-SRVEXERR', Sequence.from([400, `Could not find step with stepId ${options.stepId}`]));
```

Comment:
ds-utils.sjs can be used for this, though it needs a throwNotFound function for handling a 400.